### DISCUSSION TITLE
No ludusavi GUI appears when 2 monitors are active in Extend mode

### DISCUSSION THREAD

**Discussion by r6d2** [mtkennerly/ludusavi#396] [Q&A]
I have a PC setup with 2 monitors, generally running in "Extend" mode.

**Ludusavi** runs OK when using "PC screen only" or "Duplicate" modes, however no GUI appears when running in "Extended mode" or in "Second screen only".

Checked the task manager and _ludusavi.exe_ does not start (or fails immediately) when using the "Extend" or "Second screen only" modes.

Any ideas? Any help is appreciated. Many thanks in advance.

**Comment by mtkennerly** [comment]
Hi! I'm on Windows 11, and I'm not able to reproduce this, even when the second monitor is on a different resolution. Not sure what's going on here.

Could you open Command Prompt and try running this? And then share a copy of `%APPDATA%\ludusavi\ludusavi_rCURRENT.log` and the output in Command Prompt (if any):

```
set RUST_LOG=trace
set RUST_BACKTRACE=full
C:\path\to\ludusavi.exe
```

**Reply by r6d2** [reply]
Sure, here you go:
![image](https://github.com/user-attachments/assets/8248eb2b-be94-4be6-970f-070340b99f6c)
[ludusavi_rCURRENT.log](https://github.com/user-attachments/files/17253504/ludusavi_rCURRENT.log)

EDIT: just noticed that this _ludusavi_rCURRENT.log_ is a few days old. Tried removing it and running the command again, but it din't generate a new file with that name anywhere.

(Tried in both single monitor and dual monitor Extend mode.)🤷‍♂️

**Reply by mtkennerly** [reply]
Since that's PowerShell, the syntax is different than Command Prompt. Please try:

```
$env:RUST_LOG="trace"
$env:RUST_BACKTRACE="full"
ludusavi
```

**Reply by r6d2** [reply]
Sorry, my bad🤦‍♂️. Here you go again:
![image](https://github.com/user-attachments/assets/65fcd512-8148-4c80-89e7-44a50c59aa21)
[ludusavi_rCURRENT.log](https://github.com/user-attachments/files/17263918/ludusavi_rCURRENT.log)

**Reply by mtkennerly** [reply]
Thanks! Looks like it's rendering with Vulkan and getting an error:

```
[2024-10-04T21:19:09.086Z] WARN [wgpu_hal::vulkan] Unrecognized device error ERROR_FORMAT_NOT_SUPPORTED
```

Not sure if it's an issue with your graphics drivers or in the Rust library, but there are a couple of workarounds you can try:

* Prioritize DirectX over Vulkan: `$env:WGPU_BACKEND="dx12"`
* Switch to software rendering: `$env:ICED_BACKEND="tiny-skia"`

Could you also try the latest development build to see if it works any better?

> [ludusavi-v0.25.0-post.16+1405ae7-win64.zip](https://github.com/user-attachments/files/17264193/ludusavi-v0.25.0-post.16%2B1405ae7-win64.zip)

**Reply by mtkennerly** [reply]
Hmm, I see errors for DirectX as well, so the software renderer might be the way to go. You could also try updating your graphics drivers to see if that helps.

```
[2024-10-04T21:19:08.988Z] WARN [wgpu_hal::dx12::adapter] Device creation failed: 0x887A0020
[2024-10-04T21:19:08.995Z] WARN [wgpu_hal::dx12::adapter] Device creation failed: 0x887A0020
```

**Reply by r6d2** [reply]
Thanks, @mtkennerly, both DX12 and software rendering worked with the new development build. The GUI appears now.

PD: my AMD drivers are up to date.
